### PR TITLE
Adding groa user ratings

### PR DESCRIPTION
**Pull request by briannakeune**
# Description

Currently deployed at [dev.groa.us](dev.groa.us)

Adds the functionality to add ratings to the **user_groa_ratings** table. 

All adding of user data refactored to us **user_groa_** tables except **user_letterboxd_watched**, which has no corresponding **user_groa** table currently

Check the [docs](https://dev.groa.us/docs) to see how to use this new endpoint!

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Change Status

- [x] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

- [x] tested that this route is successful with the expected body sent.

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

File: uploading/groa-user-router.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  const newRating = {
+    date: new Date(),
+    name: req.body.name,
+    year: Number(req.body.year),
+    rating: req.body.rating * 1,
+    user_id: Number(req.params.user_id)
+  };
+  addRating(newRating)
+    .then(rated => {
+      res.status(201).json(rated);
```
Comment (cmruss) [inline uploading/groa-user-router.js]: See below

File: uploading/groa-user-router.test.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+const db = require("../database/dbConfig.js");
+const request = require("supertest");
+const server = require("../api/server.js");
+
+beforeEach(async function() {
+  await db.raw("TRUNCATE users RESTART IDENTITY CASCADE");
+});
+afterEach(async function() {
+  await db.raw("TRUNCATE user_letterboxd_ratings RESTART IDENTITY CASCADE");
```
Comment (cmruss) [inline uploading/groa-user-router.test.js]: This should be user_groa_ratings I believe

File: uploading/groa-user-router.test.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+describe("POST /users/:user_id/add-movie-rating", () => {
+  it("adds a new movie rating and gives a success message", async () => {
+    await db.seed.run();
+    const res = await request(server)
+      .post("/api/users/1/add-movie-rating")
+      .send({
+        name: "American Psycho",
+        year: 2000,
+        rating: 3
```
Comment (cmruss) [inline uploading/groa-user-router.test.js]: Needs an id?

File: uploading/models/user_groa_tables/ratings.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+};
+
+/**
+ * Inserts a rating to the database for the Groa site
+ * @param {object} rating - date, name, year, rating, user_id
+ * @returns {ratedMovie}
+ */
+async function addRating(rating) {
+  await db("user_groa_ratings").insert(rating);
+  return { message: "Success!" };
```
Comment (cmruss) [inline uploading/models/user_groa_tables/ratings.js]: In the router it looks as though we are passing the rating back to the user, but here we're not. Do we want to?

Elsewise, this should be an upsert and check if the rating exists in the table before putting it in here, unless the database has constraints on the table to only allow one movie id per user id.

The above point leads me to wonder if we don't want all user data in one table? If you already reviewed something on letterboxd, or imdb that won't show up here, allowing duplicate, triplicate ratings..

File: uploading/groa-user-router.test.js (hunk 3 above)
Comment (briannakeune) [reply uploading/groa-user-router.test.js]: Do I need to make an id? Does this table not auto increment?

File: uploading/models/user_groa_tables/ratings.js (hunk 4 above)
Comment (briannakeune) [reply uploading/models/user_groa_tables/ratings.js]: ^^^ I'll go through and clean up before I make it ready to review this morning. And we should condense the tables.

Comment (cmruss) [reply uploading/models/user_groa_tables/ratings.js]: Upsert is now functional, as is returning the rating after successful addition. Will condense the tables next.

**Comment by jgmuchiri** [comment]
🚀  Great conversation around the PR

File: uploading/groa-user-router.test.js (hunk 3 above)
Comment (cmruss) [reply uploading/groa-user-router.test.js]: I was thinking user_id, anyway, test passes

**Comment by bgingrich** [comment]
can you re-commit and gitignore docs? there's like 100 docs files that want reviewed.

**Comment by cmruss** [comment]
> can you re-commit and gitignore docs? there's like 100 docs files that want reviewed.

Yeah, that was a concern. As Brianna commented earlier, this is the only time it should be this huge, but we do want it on github.

**Pull request merged**